client/boot/project/wordpress-com.js

```diff
@@ -42,131 +30,49 @@ var config = require( 'config' ),
 import { setNextLayoutFocus, activateNextLayoutFocus } from 'state/ui/layout-focus/actions';
 
-function init() {
-	var i18nLocaleStringsObject = null;
-
-	debug( 'Starting Calypso. Let\'s do this.' );
+export function boot( currentUser, reduxStore ) {
+	debug( 'Booting WordPress.com project.' );
 
 	// prune sync-handler records more than two days old
 	syncHandler.pruneStaleRecords( '2 days' );
 
-	// Initialize i18n
-	if ( window.i18nLocaleStrings ) {
-		i18nLocaleStringsObject = JSON.parse( window.i18nLocaleStrings );
-		i18n.setLocale( i18nLocaleStringsObject );
-	}
-
-	ReactClass.injection.injectMixin( i18n.mixin );
-
-	// Infer touch screen by checking if device supports touch events
-	// See touch-detect/README.md
-	if ( touchDetect.hasTouch() ) {
-		document.documentElement.classList.add( 'touch' );
-	} else {
-		document.documentElement.classList.add( 'notouch' );
-	}
-
-	// Add accessible-focus listener
-	accessibleFocus();
-}
-
-function setUpContext( reduxStore ) {
-	page( '*', function( context, next ) {
-		var parsed = url.parse( location.href, true );
-
-		context.store = reduxStore;
-
-		// Break routing and do full page load for logout link in /me
-		if ( context.pathname === '/wp-login.php' ) {
-			window.location.href = context.path;
-			return;
-		}
-
-		// set `context.query`
-		const querystringStart = context.canonicalPath.indexOf( '?' );
-		if ( querystringStart !== -1 ) {
-			context.query = qs.parse( context.canonicalPath.substring( querystringStart + 1 ) );
-		} else {
-			context.query = {};
-		}
-		context.prevPath = parsed.path === context.path ? false : parsed.path;
-
-		// set `context.hash` (we have to parse manually)
-		if ( parsed.hash && parsed.hash.length > 1 ) {
-			try {
-				context.hash = qs.parse( parsed.hash.substring( 1 ) );
-			} catch ( e ) {
-				debug( 'failed to query-string parse `location.hash`', e );
-				context.hash = {};
-			}
-		} else {
-			context.hash = {};
-		}
-		next();
-	} );
-}
-
-function loadDevModulesAndBoot() {
-	if ( process.env.NODE_ENV === 'development' && config.isEnabled( 'render-visualizer' ) ) {
-		// Use Webpack's code splitting feature to put the render visualizer in a separate fragment.
-		// This way it won't get downloaded unless this feature is enabled.
-		// Since loading this fragment is asynchronous and we need to inject this mixin into all React classes,
-		// we have to wait for it to load before proceeding with the application's startup.
-		require.ensure( [], function() {
-			ReactClass.injection.injectMixin( require( 'lib/mixins/render-visualizer' ) );
-			boot();
-		}, 'devmodules' );
-
-		return;
-	}
-
-	boot();
-}
-
-function boot() {
-	var localeSlug;
-
-	init();
+	let localeSlug;
 
 	// When the user is bootstrapped, we also bootstrap the
 	// locale strings
 	if ( ! config( 'wpcom_user_bootstrap' ) ) {
-		localeSlug = user.get().localeSlug;
+		localeSlug = currentUser.get().localeSlug;
```

Comment:
Yes, and I think we can get rid of the part
```js
currentUser.on( 'change', function() { ... } );
```

because we already fire boot again whenever user object changes.